app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -481,44 +489,106 @@ private void handleConnectivityChange() {
-        Log.d(LOGTAG, "======> Fatal: " +
-                intent.getBooleanExtra(GeckoRuntime.EXTRA_CRASH_FATAL, false));
-
+    private void checkForCrash() {
+        final ArrayList<String> files = CrashReporterService.findCrashFiles(getBaseContext());
+        if (files.isEmpty()) {
+            Log.d(LOGTAG, "No crash files found.");
+            return;
+        }
         boolean isCrashReportingEnabled = SettingsStore.getInstance(this).isCrashReportingEnabled();
         if (isCrashReportingEnabled) {
-            sendCrashData(intent);
-
+            postCrashFiles(files);
         } else {
             if (mCrashDialog == null) {
                 mCrashDialog = new CrashDialogWidget(this);
-                mCrashDialog.setCrashDialogDelegate(() -> sendCrashData(intent));
             }
+            mCrashDialog.setCrashDialogDelegate(
+                    new CrashDialogWidget.CrashDialogDelegate() {
+                        @Override
+                        public void onSendData() {
+                            postCrashFiles(files);
+                        }
 
+                        @Override
+                        public void onDoNotSendData() {
+                            for (String file : files) {
+                                Log.e(LOGTAG, "Deleting crashfile: " + file);
+                                getBaseContext().deleteFile(file);
+                            }
+                        }
+                    }
+            );
             mCrashDialog.show(UIWidget.REQUEST_FOCUS);
         }
     }
 
-    private void sendCrashData(final Intent intent) {
+    private void handleContentCrashIntent(@NonNull final Intent intent) {
+        Log.e(LOGTAG, "Got content crashed intent");
+        final String dumpFile = intent.getStringExtra(GeckoRuntime.EXTRA_MINIDUMP_PATH);
+        final String extraFile = intent.getStringExtra(GeckoRuntime.EXTRA_EXTRAS_PATH);
+        Log.d(LOGTAG, "Dump File: " + dumpFile);
+        Log.d(LOGTAG, "Extras File: " + extraFile);
+        Log.d(LOGTAG, "Fatal: " + intent.getBooleanExtra(GeckoRuntime.EXTRA_CRASH_FATAL, false));
+
+        boolean isCrashReportingEnabled = SettingsStore.getInstance(this).isCrashReportingEnabled();
+        if (isCrashReportingEnabled) {
+            postCrashFiles(dumpFile, extraFile);
+        } else {
+            if (mCrashDialog == null) {
+                mCrashDialog = new CrashDialogWidget(this);
+            }
+            mCrashDialog.setCrashDialogDelegate(() -> postCrashFiles(dumpFile, extraFile));
+            mCrashDialog.show(UIWidget.REQUEST_FOCUS);
+        }
+    }
+
+    private void sendCrashFiles(@NonNull final String aDumpFile, @NonNull final String aExtraFile) {
+        try {
+            GeckoResult<String> result = CrashReporter.sendCrashReport(VRBrowserActivity.this, new File(aDumpFile), new File(aExtraFile), getString(R.string.crash_app_name));
+
+            result.then(crashID -> {
```

Comment:
You can use `result.accept` here which is a little nicer.

Reply:
Thanks, done.